source/extensions/filters/udp/udp_proxy/session_filters/dynamic_forward_proxy/proxy_filter.cc

```diff
@@ -49,6 +49,12 @@ ReadFilterStatus ProxyFilter::onNewSession() {
     host = host_filter_state->asString();
   }
 
+  if (host.empty()) {
+    // The session may be routed to a non-dynamic cluster.
+    load_dns_cache_completed_ = true;
```

Comment:
Can you describe what would happen after early return here given cache_load_handle_ will be left as nullptr?

Reply:
Not used (It's only used in `onNewSession` and because we return early, it won't be used at all).
Also, this aligns with the implementation of the `sni dynamic forward proxy` filter for tcp: https://github.com/envoyproxy/envoy/blob/3a1a72e9d7a272e452bb5499c4dcdc43e1eee5ed/source/extensions/filters/network/sni_dynamic_forward_proxy/proxy_filter.cc#L50-L52